sdk/storage/Azure.Storage.DataMovement.Files.Shares/tests/ShareFileResourceTests.cs

```diff
@@ -0,0 +1,449 @@
+                new ShareClientOptions());
+            ShareFileStorageResource sourceResource = new ShareFileStorageResource(mockSource.Object);
+
+            Mock<ShareFileClient> mockDestination = new(
+                new Uri("https://storageaccount.file.core.windows.net/container/destinationfile"),
+                new ShareClientOptions());
+
+            mockDestination.Setup(b => b.UploadRangeFromUriAsync(It.IsAny<Uri>(), It.IsAny<HttpRange>(), It.IsAny<HttpRange>(), It.IsAny<ShareFileUploadRangeFromUriOptions>(), It.IsAny<CancellationToken>()))
+                .Throws(new RequestFailedException(status: 404, message: "The specified resource does not exist.", errorCode: "ResourceNotFound", default));
+            ShareFileStorageResource destinationResource = new ShareFileStorageResource(mockDestination.Object);
+
+            // Act
+            int length = 1024;
+            await TestHelper.AssertExpectedExceptionAsync<RequestFailedException>(
+                destinationResource.CopyFromUriInternalAsync(sourceResource, false, length),
+                e =>
+                {
+                    Assert.AreEqual("ResourceNotFound", e.ErrorCode);
+                });
+        }
+
+        [Test]
+        public async Task CopyBlockFromUriAsync()
+        {
+            // Arrange
+            Mock<ShareFileClient> mockSource = new(
+                new Uri("https://storageaccount.file.core.windows.net/container/sourcefile"),
+                new ShareClientOptions());
+            ShareFileStorageResource sourceResource = new ShareFileStorageResource(mockSource.Object);
+
+            Mock<ShareFileClient> mockDestination = new(
+                new Uri("https://storageaccount.file.core.windows.net/container/destinationfile"),
+                new ShareClientOptions());
+
+            mockDestination.Setup(b => b.UploadRangeFromUriAsync(It.IsAny<Uri>(), It.IsAny<HttpRange>(), It.IsAny<HttpRange>(), It.IsAny<ShareFileUploadRangeFromUriOptions>(), It.IsAny<CancellationToken>()))
+                .Returns(Task.FromResult(Response.FromValue(
+                    ShareModelFactory.ShareFileUploadInfo(
+                        eTag: new ETag("eTag"),
+                        lastModified: DateTimeOffset.UtcNow,
+                        contentHash: default,
+                        isServerEncrypted: false),
+                    new MockResponse(200))));
+            ShareFileStorageResource destinationResource = new ShareFileStorageResource(mockDestination.Object);
+
+            int length = 1024;
+            await destinationResource.CopyFromUriInternalAsync(sourceResource, false, length);
+
+            // Act
+            await destinationResource.CopyBlockFromUriInternalAsync(
+                sourceResource: sourceResource,
+                overwrite: false,
+                range: new HttpRange(0, length),
+                completeLength: length);
+        }
```

Comment:
Should be verifying the mock destination client.

Reply:
done